cs-abbyy/src/main/java/io/cloudslang/content/abbyy/services/ProcessImageService.java

```diff
@@ -0,0 +1,266 @@
+        }
+
+        if (input.getImageSource() != null) {
+            urlBuilder.addParameter(ConnectionConstants.QueryParams.IMAGE_SOURCE, input.getImageSource().toString());
+        }
+
+        urlBuilder.addParameter(ConnectionConstants.QueryParams.CORRECT_ORIENTATION, String.valueOf(input.isCorrectOrientation()));
+
+        urlBuilder.addParameter(ConnectionConstants.QueryParams.CORRECT_SKEW, String.valueOf(input.isCorrectSkew()));
+
+        if (input.isReadBarcodes() != null) {
+            urlBuilder.addParameter(ConnectionConstants.QueryParams.READ_BARCODES, String.valueOf(input.isReadBarcodes()));
+        }
+
+        if (!input.getExportFormats().isEmpty()) {
+            urlBuilder.addParameter(ConnectionConstants.QueryParams.EXPORT_FORMAT, StringUtils.join(input.getExportFormats(), ','));
+        }
+
+        if (StringUtils.isNotEmpty(input.getDescription())) {
+            urlBuilder.addParameter(ConnectionConstants.QueryParams.DESCRIPTION, input.getDescription());
+        }
+
+        if (StringUtils.isNotEmpty(input.getPdfPassword())) {
+            urlBuilder.addParameter(ConnectionConstants.QueryParams.PDF_PASSWORD, input.getPdfPassword());
+        }
+
+        if (input.getExportFormats().contains(ExportFormat.XML)) {
+            urlBuilder.addParameter(ConnectionConstants.QueryParams.WRITE_FORMATTING, String.valueOf(input.isWriteFormatting()))
+                    .addParameter(ConnectionConstants.QueryParams.WRITE_RECOGNITION_VARIANTS, String.valueOf(input.isWriteRecognitionVariants()));
+        }
+
+        if (input.getExportFormats().contains(ExportFormat.PDF_SEARCHABLE)) {
+            urlBuilder.addParameter(ConnectionConstants.QueryParams.WRITE_TAGS, input.getWriteTags().toString());
+        }
+
+        return urlBuilder.build().toString();
+    }
+
+
+    @Override
+    protected void handleTaskCompleted(ProcessImageInput request, AbbyyResponse response, Map<String, String> results) throws Exception {
+        super.handleTaskCompleted(request, response, results);
+
+        Map<ExportFormat, String> failure = new HashMap<>();
+
+        for (ExportFormat exportFormat : request.getExportFormats()) {
+            try {
+                switch (exportFormat) {
+                    case TXT:
+                        String txt = getProcessingResult(request, response, exportFormat);
+                        results.put(OutputNames.TXT_RESULT, txt);
+                        if (request.getDestinationFile() != null) {
+                            saveClearTextResultOnDisk(request, txt, exportFormat);
+                        }
+                        break;
+                    case XML:
+                        String xml = getProcessingResult(request, response, exportFormat);
+                        AbbyySdkException validationEx = resultValidator.validate(xml);
+                        if (validationEx != null) {
+                            throw validationEx;
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Exception [AbbyySdkException] is explicitly thrown on this line but is caught in the outer catch clause with generic exception on line [162].  Consider adding a catch clause of this specific exception in the outer try statement.